src/scripts/items/comparison/perfectionScore.ts

```diff
@@ -98,15 +101,13 @@ export function perfectionScore(item: Item) {
     });
   }
 
-  // % enhanced defense armor always spawn with max def + 1
   if (
     (item.quality === ItemQuality.UNIQUE || item.quality === ItemQuality.SET) &&
-    "def" in base &&
-    !ranges.some(({ prop }) => prop === "ac%")
```

Comment:
I'm confused by this fix. If an item has ac%, then the defense **isn't** a range, so it shouldn't be included in the perfection score at all. Which is what the previous code was doing, so `addProp()` was never called for items with ac%.

Can you please describe the bug before, which items had an issue? Or link the PR to an issue if there is one already.

Reply:
This fixes https://github.com/youdz/d2-stash-organizer/issues/15
The Problem is easy to see with the Milabrega Set. Armors and Shields are always perfect but have a perfection score of 106% and 150%. The check of ranges contains "ac%" isn't right because the items themselves don't have the attribute but gain it through the setbonus. I think the problem also occurs with items that have "ac" as a property but I didn't check that in detail. Instead of listing all attributes that can cause this bug I made the solution independent from the causing properties and fixed the calculation formula. I think this is a more robust approach as it doesn't depend on knowing which attributes cause the problem.

Before it was
`score = (nbProps * score + (value - min) / (max - min)) / ++nbProps;`
for milabrega's orb: value = 19, min = 16, max = 18 which results in 1,5 because value is 1 above max

Now if the value is higher than max (should only be by one in these special cases) through Math.min(value, max) it uses max instead which results in a correct calculation.

Reply:
>This fixes #15 (see the title of this PR 😇)

🤦‍♂️ 

Anyway, I see the issue. Thanks for the investigation. I believe the more robust fix is to change the custom logic for sets and instead make sure `ranges` contains set properties too. That also results in less code duplication, which I like. I tried it quickly and both broken pieces of Milabrega now work correctly. This also made me realize that plain ac bonus isn't displayed correctly in the tooltip.

Give me a minute, I'll use your PR as a base so you're still the author, and update it with my preference. I am not that fond of the `Math.min` solution because it makes computation bugs harder to fix, actually. If a value is above the expected range, I kind of like that it looks weird and someone will report it so I can investigate.